pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,139 @@
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func filterFindingsBySeverity(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return utils.Contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func filterFindings(findingsModel local_models.LocalFinding, filter []FindingsFilterFunc) local_models.LocalFinding {
+	filteredFindings := local_models.LocalFinding{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filter {
+			if match {
+				match = filter(finding)
+			}
+		}
+		if match {
+			filteredFindings.Findings = append(filteredFindings.Findings, finding)
+		}
+	}
+	return filteredFindings
+}
+
+func filterFindingsEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	ff_transform_enabled := config.GetBool(configuration.FF_TRANSFORMATION_WORKFLOW)
+	output = []workflow.Data{}
+
+	logger.Println("Workflow Name:", FilterFindingsWorkflowName)
+
+	if !ff_transform_enabled {
+		return input, nil
+	}
+
+	var findingsModels []local_models.LocalFinding
+
+	for _, data := range input {
+		if strings.HasPrefix(data.GetContentType(), content_type.LOCAL_FINDING_MODEL) {
+			var findingsModel local_models.LocalFinding
+			findingsBytes, ok := data.GetPayload().([]byte)
+			if !ok {
+				var findingsError snyk_errors.Error = snyk_errors.Error{
+					Title:          "Invalid Payload Type",
+					Classification: "Internal",
+					Level:          "warning",
+					Detail:         "Invalid payload type",
+				}
+				logger.Err(findingsError).Msg(findingsError.Error())
+				return nil, findingsError
+			}
+			err := json.Unmarshal(findingsBytes, &findingsModel)
+			if err != nil {
+				var unmarshallError snyk_errors.Error = snyk_errors.Error{
+					Title:          "Failed to unmarshall findings",
+					Classification: "Internal",
+					Level:          "warning",
+					Detail:         err.Error(),
+				}
+				logger.Err(unmarshallError).Msg(unmarshallError.Error())
+				return nil, unmarshallError
+			}
+			findingsModels = append(findingsModels, findingsModel)
+		} else {
+			output = append(output, data)
+		}
+	}
+	if len(findingsModels) == 0 {
+		return output, nil
+	}
+	for _, findingsModel := range findingsModels {
+		severityOrder := findingsModel.Summary.SeverityOrderAsc
+		filter := filterFindingsBySeverity(config.GetString(configuration.FLAG_SEVERITY_THRESHOLD), severityOrder)
```

Comment:
Suggestion: If the severity-threshold is not set, there will be a filter with all severities. This seems unnecessary effort. What I want to say is, we don't need to filter of no filter parameter is specified.